src/main/scala/com/amazon/deequ/constraints/Constraint.scala

```diff
@@ -897,3 +898,29 @@ object Constraint {
     }
 
 }
+
+/**
+ * Data Synchronization Constraint
+ * @param analyzer Data Synchronization Analyzer
+ * @param hint hint
+ */
+case class DataSynchronizationConstraint(analyzer: DataSynchronizationAnalyzer, hint: Option[String])
+  extends Constraint {
+
+  override def evaluate(metrics: Map[Analyzer[_, Metric[_]], Metric[_]]): ConstraintResult = {
+
+    val anz = Try(metrics.filter(i => i._1.isInstanceOf[DataSynchronizationAnalyzer]).head._2)
```

Comment:
Can we write it as the following, for better readability?

```
val (_, anz) = Try(metrics.filter { case(analyzer, _)  => analyzer.isInstanceOf[DataSynchronizationAnalyzer] }.head)
```

What happens if `.head` is called on empty list? Is that possible?

Reply:
will refactor to make it improve readability, thanks for suggestion.